contrib/database/sql/conn.go

```diff
@@ -114,3 +120,47 @@ func (tc *tracedConn) QueryContext(ctx context.Context, query string, args []dri
 	tc.tryTrace(ctx, "Query", query, start, err)
 	return rows, err
 }
+
+// traceParams stores all information related to tracing the driver.Conn
+type traceParams struct {
+	rc         *registerConfig
+	oc         *openConfig
+	driverName string
+	resource   string
+	meta       map[string]string
+}
+
+// tryTrace will create a span using the given arguments, but will act as a no-op when err is driver.ErrSkip.
+func (tp *traceParams) tryTrace(ctx context.Context, resource string, query string, startTime time.Time, err error) {
+	if err == driver.ErrSkip {
+		// Not a user error: driver is telling sql package that an
+		// optional interface method is not implemented. There is
+		// nothing to trace here.
+		// See: https://github.com/DataDog/dd-trace-go/issues/270
+		return
+	}
+	name := fmt.Sprintf("%s.query", tp.driverName)
+	opts := []ddtrace.StartSpanOption{
+		tracer.SpanType(ext.SpanTypeSQL),
+		tracer.StartTime(startTime),
+	}
+	if tp.oc.serviceName != "" {
```

Comment:
This could be avoided if we just merged the `Open` config into the `Register` config upon opening.

Reply:
This has been updated, and config is merged at an earlier point.